packages/wallet-adapter-react/src/WalletProvider.tsx

```diff
@@ -169,21 +172,9 @@ export const AptosWalletAdapterProvider: FC<AptosWalletProviderProps> = ({
     });
   }, [connected]);
 
-  // Handle the adapter's ready state change event
-  const handleReadyStateChange = (wallet: Wallet) => {
-    setWallets((prevWallets) => {
-      const index = prevWallets.findIndex(
-        (currWallet) => currWallet === wallet
-      );
-      if (index === -1) return prevWallets;
-
-      return [
-        ...prevWallets.slice(0, index),
-        wallet,
-        ...prevWallets.slice(index + 1),
-      ];
-    });
-  };
+  // Whenever the readyState of any supported wallet changes which we force a rerender.
+  // See https://github.com/aptos-labs/aptos-wallet-adapter/pull/129#issuecomment-1519026572 for reasoning.
+  const handleReadyStateChange = (wallet: Wallet) => forceUpdate();
```

Comment:
can we get this change in a separate PR and keep this one focused on fixing the autoconnect?